### PR TITLE
bump Sargon to 1.0.1

### PR DESCRIPTION
**Pull request by CyonAlexRDX**
# Sargon 1.0.1 
This contains the 10x speedup to JSON Decoding and analysis of Profile
Contents and encryption/decryption, part of https://github.com/radixdlt/sargon/pull/144

>[!NOTE]
> Deprecations: I've deprecated the `Data` based API's or Profile in Swift Sargon since they are just convenience methods/functions which wrap `String.init:data:encoding)` and `String.data:encoding` utf8 roundtrips. So whenever we CAN we SHOULD use JSON `String` based API's directly, for maximum performance (i.e. when we add Persona avatars to Profile and it is several megabytes we might save some milliseconds avoiding unnessary `utf8` conversions).

> [!TIP]
> Enable Rust logging by default - it is **VERY verbose** you might wanna comment it out from WalletApp.swift

image

### CODE REVIEW

File: RadixWallet.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -8607,15 +8606,15 @@
 			repositoryURL = "https://github.com/pointfreeco/swift-identified-collections";
 			requirement = {
 				kind = exactVersion;
-				version = 1.0.1;
+				version = 1.0.2;
```
Comment (Sajjon) [inline RadixWallet.xcodeproj/project.pbxproj:8609]: I bumped IdentifiedCollections to `1.0.2` as well, containing a bug fix of of `sort` crashing (Gustaf already changed our TX history to sort using Array instead to circumvent this)

File: RadixWallet.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 2]
```diff
@@ -114,8 +114,8 @@
       "kind" : "remoteSourceControl",
       "location" : "https://github.com/radixdlt/sargon",
       "state" : {
-        "revision" : "30cfe39a9e108ac2e9c7a4f01fb917ba131869a1",
-        "version" : "0.7.19"
+        "revision" : "60006952864e2b4fb1e511a83844694819132d04",
+        "version" : "1.0.1"
```
Comment (Sajjon) [inline RadixWallet.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:118]: 🎉 I chose version `1.0.x` of Sargon to mark the first wallet release to use Sargon Profile 🥳

File: RadixWallet/Features/ProfileBackupsFeature/Shared/ExportableProfileFile.swift [hunk 3]
```diff
@@ -44,7 +44,7 @@ extension ExportableProfileFile {
 	}
 
 	public init(data: Data) throws {
-		switch Profile.analyzeFile(contents: data) {
+		switch Profile.analyzeContents(data: data) {
```
Comment (Sajjon) [inline RadixWallet/Features/ProfileBackupsFeature/Shared/ExportableProfileFile.swift:47]: this was the only breaking change...

File: RadixWallet/WalletApp.swift [hunk 4]
```diff
@@ -6,6 +6,15 @@ import SwiftUI
 struct WalletApp: SwiftUI.App {
 	@UIApplicationDelegateAdaptor var delegate: AppDelegate
 
+	init() {
+		#if DEBUG
+		// MUST NOT be called twice (panics...)
+		// MUST NOT be enabled in PROD (not ensured that secrets are omitted)
+		// This is VERY VERY verbose, so mosly useful for debugging.
+		enableLoggingFromRust()
```
Comment (Sajjon) [inline RadixWallet/WalletApp.swift:14]: VERY verbose, we might wanna disable it by default, but lets try this for now... see here, how verbose: https://github.com/radixdlt/babylon-wallet-ios/pull/1113

**Review by kugel3 — APPROVED** [review]

**Pull request merged**